Jenkinsfile

```diff
@@ -21,6 +21,42 @@ void setBuildStatus(String message, String state) {
     ]);
 }
 
+def runBuild(String name, String nodeLabel) {
+    stage("Building " + name + " (" + env.NODE_NAME + ")") {
+        dir(name) {
+            // Delete CMake cache folder
+            dir("build") {
+                deleteDir();
+            }
+
+            // Generate unique name for message
+            def uniqueMsg = "msg_" + name + "_" + env.NODE_NAME;
+
+            setBuildStatus("Building " + name, "PENDING");
+
+            // Build tests and set build status based on return code
+            def statusCode = sh script:"./build_all.sh 1> \"" + uniqueMsg + "\" 2> \"" + uniqueMsg + "\"", returnStatus:true
+            if(statusCode != 0) {
+                setBuildStatus("Building " + name, "FAILURE");
+            }
+
+            // Parse test output for GCC warnings
+            // **NOTE** driving WarningsPublisher from pipeline is entirely undocumented
+            // this is based mostly on examples here https://github.com/kitconcept/jenkins-pipeline-examples
+            // **YUCK** fatal errors aren't detected by the 'GNU Make + GNU C Compiler (gcc)' parser
+            // however JENKINS-18081 fixes this for
+            // the 'GNU compiler 4 (gcc)' parser at the expense of it not detecting make errors...
+            def parserName = ("mac" in nodeLabel) ? "Apple LLVM Compiler (Clang)" : "GNU compiler 4 (gcc)";
+            step([$class: "WarningsPublisher",
```

Comment:
It might be worth following this sorry tale and trying out "GNU Make + GNU C Compiler (gcc)". However, the warnings plugin is deprecated and shit so you _could_ do me a **massive** favour and have a go with the snazzily named "Warnings Next Generation" plugin 😄 It is installed on the jenkins machine and documented [here](https://www.jenkins.io/doc/pipeline/steps/warnings-ng/). I would guesstimate that the ``scanForIssues`` command is what you want.

Reply:
I tried this and it *seemed* to work (I deliberately put a bug in a file and Jenkins detected it) but now I seem to be getting this warning:

```

java.lang.IllegalStateException: ID gcc is already used by another action: io.jenkins.plugins.analysis.core.model.ResultAction for GNU C Compiler (gcc)

	at io.jenkins.plugins.analysis.core.steps.IssuesPublisher.ensureThatIdIsUnique(IssuesPublisher.java:117)
	at io.jenkins.plugins.analysis.core.steps.IssuesPublisher.attachAction(IssuesPublisher.java:92)
	at io.jenkins.plugins.analysis.core.steps.IssuesRecorder.publishResult(IssuesRecorder.java:623)
	at io.jenkins.plugins.analysis.core.steps.IssuesRecorder.record(IssuesRecorder.java:556)
	at io.jenkins.plugins.analysis.core.steps.IssuesRecorder.perform(IssuesRecorder.java:521)
	at io.jenkins.plugins.analysis.core.steps.RecordIssuesStep$Execution.run(RecordIssuesStep.java:942)
	at io.jenkins.plugins.analysis.core.steps.RecordIssuesStep$Execution.run(RecordIssuesStep.java:907)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Any ideas?

Reply:
Maybe try turning off ``aggregatingResults`` as you only use a single tool (based on https://issues.jenkins.io/browse/JENKINS-54027)

Reply:
you may also be able to set ``id: env.NODE_NAME`` and get actual per-node error reports rather than the weird blend we have on GeNN

Reply:
YESSSSS! Setting the ``id`` totally solved it.

I've tried deliberately breaking tests and adding warnings and it all seems to work :partying_face: